shell/app/config-page/components/chart-dashboard/chart-dashboard.tsx

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React from 'react';
+import { isEmpty } from 'lodash';
+import { PureBoardGrid, Holder } from 'common';
+
+const DashBoard = React.memo(PureBoardGrid);
+
+const ChartDashboard = (props: CP_CHART_DASHBOARD.Props) => {
+  const { props: configProps, state } = props;
+  const { layout } = configProps || {};
+  return (
+    <Holder when={isEmpty(layout)}>
```

Comment:
is it possible to remove `isEmpty`

Reply:
Seemingly impossible, layout could be {} / null , dependence api response